ark/phenotyping/run_trained_som.R

```diff
@@ -44,31 +44,32 @@ markers <- colnames(somWeights)
 print("Mapping data to cluster labels")
 for (i in 1:length(fovs)) {
     # read in pixel data
-    fileName <- paste(fovs[i], ".feather", sep="")
-    matPath <- paste(pixelMatDir, fileName, sep="/")
-    fovPixelData <- as.matrix(arrow::read_feather(matPath, col_select=all_of(markers)))
+    fileName <- file.path(fovs[i], "feather", fsep=".")
+    matPath <- file.path(pixelMatDir, fileName)
+    fovPixelData <- arrow::read_feather(matPath)
 
     # 99.9% normalize pixel data
     for (marker in markers) {
         # this prevents all- or mostly-zero columns from getting normalized and becoming NA/Inf
         if (normVals[1, marker] != 0) {
-            fovPixelData[, marker] = fovPixelData[, marker] / normVals[1, marker]
+            fovPixelData[, marker] <- fovPixelData[, marker] / normVals[1, marker]
         }
     }
 
     # map FlowSOM data
-    clusters <- FlowSOM:::MapDataToCodes(somWeights, fovPixelData)
+    clusters <- FlowSOM:::MapDataToCodes(somWeights, as.matrix(fovPixelData[, markers]))
 
     # assign cluster labels column to pixel data
-    fovPixelData <- as.matrix(cbind(as.matrix(fovPixelData), cluster=clusters[,1]))
+    fovPixelData <- cbind(fovPixelData, cluster=clusters[,1])
```

Comment:
I think you said cbind was the slow part here, right? It is faster to initialize this matrix from the beginning with a blank consensus clustered column, and assign it here? Or is assignment equally slow

Reply:
Let me try using `$` assignment first; that's what Candace does in her scripts. Initializing this matrix with a blank column and assigning later would essentially be the same, just with an extra step.

Reply:
Sounds good. There's probably some best practices about fastest way to update dataframes